packages/datastore/src/datastore.ts

```diff
@@ -596,6 +637,24 @@ namespace Datastore {
     readonly type: 'datastore-transaction';
   }
 
+  /**
+   * A message of a datastore GC opportunity.
+   *
+   * You can send such a message to the datastore when you
+   * are sure that all concurrent transactions have been applied,
+   * i.e. any transactions that arrive later will have the current
+   * state as a base. This will provide  a hint to the datastore
+   * that it is safe to clean up some internal state.
```

Comment:
This note might not be correct given my misunderstanding which models we were using (causal delivery not being a requirement). Somebody will have to check:
- Is this correct? If not:
- Are there ever any time that is better for GC than others?